integrations/influxdb_connector/influxdb-timestream-connector/src/timestream_utils.rs

```diff
@@ -139,45 +154,94 @@ pub async fn database_exists(
     }
 }
 
+#[tracing::instrument(skip_all, level = tracing::Level::TRACE)]
 pub async fn ingest_records(
-    client: &timestream_write::Client,
-    database_name: &str,
-    table_name: &str,
-    records: &mut [timestream_write::types::Record],
+    client: Arc<timestream_write::Client>,
+    database_name: Arc<String>,
+    table_name: String,
+    records: Vec<timestream_write::types::Record>,
 ) -> Result<(), Error> {
     // Ingest records to Timestream in batches of 100 (Max supported Timestream batch size)
+    // in parallel
 
     let mut records_ingested: usize = 0;
     const MAX_TIMESTREAM_BATCH_SIZE: usize = 100;
 
-    let mut records_chunked: Vec<Vec<timestream_write::types::Record>> = records
-        .chunks(MAX_TIMESTREAM_BATCH_SIZE)
+    // Chunk records in parallel using rayon (par_chunks)
+    let records_chunked: Vec<Vec<timestream_write::types::Record>> = records
+        .par_chunks(MAX_TIMESTREAM_BATCH_SIZE)
         .map(|sub_records| sub_records.to_vec())
         .collect();
-    for chunk in records_chunked.iter_mut() {
-        records_ingested += chunk.len();
-        match client
-            .write_records()
-            .database_name(database_name)
-            .table_name(table_name)
-            .set_records(Some(std::mem::take(chunk)))
-            .send()
+
+    // Use a semaphore to limit the maximum number of threads used to ingest chunks in parallel
+    let semaphore = Arc::new(Semaphore::new(NUM_TIMESTREAM_INGEST_THREADS));
```

Comment:
Can you give a more descriptive variable name like `ingestion_semaphore`

Reply:
Renamed semaphores to `ingestion_semaphore`.